src/log_event_group.py

```diff
@@ -0,0 +1,30 @@
+""" Module for defining an object representation of a detected event group """
+
+class LogEventGroup:
+    """ Stores the details of a singular occurence of an event group 
+        
+        Attributes
+        ----------
+        tag: Label for the event group
+        trigger_log_events: ordered list of events that comprise
+                            an event group
+        context_log_events: list of context events
+        all_log_events: chonological order of all events"""
+
+    def __init__(self, tag, trigger_log_events=[], context_log_events=[],
+                 all_log_events=[]):
+
+        self.tag = tag
+        self.trigger_log_events = trigger_log_events
+        self.context_log_events = context_log_events
+        self.all_log_events = all_log_events
+
+    def add_log_event(self, trigger_log_event=None, context_log_event=None):
```

Comment:
Are you only supposed to specify one of these two arguments? If so it would probably be good to add a check that they aren't both set, since having the second argument get ignored is probably an error and not what the caller intended.

Reply:
The intention there was to leave the capability to add both if needed, but now I think that is better captured with separate 'add_context' and 'add_trigger' methods.

Reply:
Hmmm. Just reviewed how this was being used a little further, should be only one getting added, going with original fix.